applications/StructuralMechanicsApplication/custom_constitutive/constitutive_laws_integrators/high_cycle_fatigue_law_integrator.h

```diff
@@ -0,0 +1,322 @@
+
+// System includes
+
+// Project includes
+#include "includes/define.h"
+#include "includes/checks.h"
+#include "includes/serializer.h"
+#include "includes/properties.h"
+#include "utilities/math_utils.h"
+#include "structural_mechanics_application_variables.h"
+#include "custom_utilities/constitutive_law_utilities.h"
+
+namespace Kratos
+{
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+    // The size type definition
+    typedef std::size_t SizeType;
+    
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/**
+ * @class HighCycleFatigueLawIntegrator
+ * @ingroup StructuralMechanicsApplication
+ * @brief: This object computes all the required information for the high cycle fatigue constitutive law.
+ * @details The definitions of these classes is completely static, the derivation is done in a static way
+ * @tparam TVoigtSize Strain size
+ * @author Sergio Jimenez, Alejandro Cornejo & Lucia Barbu
+ */
+template <SizeType TVoigtSize = 6>
+class HighCycleFatigueLawIntegrator
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+
+    /// Counted pointer of HighCycleFatigueLawIntegrator
+    KRATOS_CLASS_POINTER_DEFINITION(HighCycleFatigueLawIntegrator);
+
+    /// Initialization constructor
+    HighCycleFatigueLawIntegrator()
+    {
+    }
+
+    /// Copy constructor
+    HighCycleFatigueLawIntegrator(HighCycleFatigueLawIntegrator const &rOther)
+    {
+    }
+
+    /// Assignment operator
+    HighCycleFatigueLawIntegrator &operator=(HighCycleFatigueLawIntegrator const &rOther)
+    {
+        return *this;
+    }
+
+    /// Destructor
+    virtual ~HighCycleFatigueLawIntegrator()
+    {
+    }
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    /**
+     * @brief This method integrates checks if the uniaxial stress value of the previous step was a maximum or a minimum 
+     * @param CurrentStress The signed uniaxial stress in the current step.
+     * @param MaximumStress Identifies if the stress in the previous step was a maximum (signed uniaxial stress) in the calculation. Otherwise it is 0.
+     * @param MinimumStress Identifies if the stress in the previous step was a minimum (signed uniaxial stress) in the calculation. Otherwise it is 0.
+     * @param PreviousMaximumStress Latest identified maximum (signed uniaxial stress) in the calculation.
+     * @param PreviousMinimumStress Latest identified minimum (signed uniaxial stress) in the calculation.
+     * @param PreviousStresses Vector containing the stress value (signed uniaxial stresses) from the two previous steps.
+     * @param NumberOfCycles Number of cycles for the applied load.
+     * @param CycleCounter Boolean variable used on the update operation for the SetMaxStress.
+     */
+    static void CalculateMaximumAndMinimumStresses(
+        const double CurrentStress,
+        double &rMaximumStress, 
```

Comment:
This function is quite complex. Does it belong in the public section?

Reply:
Yes, as it is called from the .cpp of the CL it has to be public. In fact, I have tried to write it as private but then the compilation fails.